backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/aries/ProofManager.java

```diff
@@ -154,6 +267,39 @@ public void handleProofEvent(PresentationExchangeRecord proof) {
                         }));
     }
 
+    // handles all proof request
+    public void handleProofRequestEvent(PresentationExchangeRecord proof) {
```

Comment:
Is this intentional that you only track requested presentations here, and then all following events with handleProof()?

Reply:
We can either have the handler determine the type of event and call a simplified handler. Or we could write one big handler to process everything with events of that type. I went with the former to keep the methods smaller.

Let me know if that works.

Reply:
As states will get more important the best option is to keep it small, but leave the decision in the webhook handler

Reply:
The naming will need some work, i also think all these event handlers should be moved into to their own files.